src/callbacks/wandb_callbacks.py

```diff
@@ -196,3 +196,40 @@ def on_validation_epoch_end(self, trainer, pl_module):
 
             self.preds.clear()
             self.targets.clear()
+            
+
+class ImagePredictionLogger(Callback):
+    """Logs a validation batch and their predictions to wandb.
+       Example adapted from:
+       https://wandb.ai/wandb/wandb-lightning/reports/Image-Classification-using-PyTorch-Lightning--VmlldzoyODk1NzY
+    """
+
+    def __init__(self, num_samples):
+        super().__init__()
+        self.num_samples = num_samples
+
+    def on_sanity_check_start(self, trainer, pl_module):
+        self.ready = False
+
+    def on_sanity_check_end(self, trainer, pl_module):
+        """Start executing this callback only after all validation sanity checks end."""
+        self.ready = True
+
+    def on_validation_epoch_end(self, trainer, pl_module):
+        # get a validation batch from the calidation dat loader
```

Comment:
You need to add here:
```
if self.ready:
```
as it is done in other callbacks. Lightning executes a couple of short sanity checks for validation dataloader at the beginning of the training, which will log extra images if you don't add this line, it also could result in crashing the run if you set more num_samples than available during sanity check.